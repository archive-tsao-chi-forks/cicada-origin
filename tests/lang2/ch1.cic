// 1. The More Things Change, the More They Stay the Same

{
  _ = "ratatouille": String
  _ = cons("ratatouille", "baguette"): Pair(String, String)
  _ = same: Equal(String, "ratatouille", "ratatouille")
  _ = same: Equal(
    Pair(String, String),
    cons("ratatouille", "baguette"),
    cons("ratatouille", "baguette"))
  _ = car(cons("ratatouille", "baguette"): Pair(String, String))
  _ = same: Equal(
    String,
    "ratatouille",
    car(cons("ratatouille", "baguette"): Pair(String, String)))
  _ = Pair(String, String): Type
  _ = String: Type
  _ = Type: Type
  _ = same: Equal(Type, String, String)
  _ = Pair(
    car(cons(String, "olive"): Pair(Type, String)),
    cdr(cons("oil", String): Pair(String, Type))): Type
  _ = cons(
    cdr(cons(String, "olive"): Pair(Type, String)),
    car(cons("oil", String): Pair(String, Type))
  ): Pair(
    car(cons(String, "olive"): Pair(Type, String)),
    cdr(cons("oil", String): Pair(String, Type)))
  _ = 172: Nat
  _ = 0: Nat
  _ = zero: Nat
  _ = same: Equal(Nat, 0, zero)
  _ = same: Equal(Nat, zero, 0)
  one = add1(0)
  one
}
