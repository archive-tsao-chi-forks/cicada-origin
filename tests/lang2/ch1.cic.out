Same: Equal(Nat, 0, 0)
